rusk/src/lib/http/rusk.rs

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use std::sync::Arc;
+
+use rusk_abi::ContractId;
+
+use crate::http::{DataType, WsRequest, WsResponse};
+use crate::Rusk;
+
+use super::event::WsTarget;
+
+impl Rusk {
+    pub(crate) async fn handle_request(
```

Comment:
Do I smell a trait? Given that this same function is implemented by `RuskNode`. Might be useful in request routing as well.